census/tf-single/trainer/task.py

```diff
@@ -0,0 +1,203 @@
+# Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+
+import argparse
+import tensorflow as tf
+import numpy as np
+import pandas as pd
+from tensorflow.contrib.layers.python.ops import sparse_feature_cross_op
+from tensorflow.contrib.layers.python.ops import bucketization_op
+
+import model
+
+#https://www.tensorflow.org/tutorials/wide_and_deep/
+#https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/layers/feature_column.py
+
+# csv columns in the input file
+CSV_COLUMNS = ('age', 'workclass', 'fnlwgt', 'education', 'education_num',
+               'marital_status', 'occupation', 'relationship', 'race', 'gender',
+               'capital_gain', 'capital_loss', 'hours_per_week', 'native_country',
+               'income_bracket')
+
+CATEGORICAL_COLS = ('gender', 'race', 'education', 'marital_status',
+                    'relationship', 'workclass','occupation', 'native_country')
+
+CONTINUOUS_COLS = ('age', 'education_num', 'capital_gain', 'capital_loss', 'hours_per_week')
+
+LABEL_COL = 'income_bracket'
+
+def read_input_data(file_name, skiprows=None):
+  """Read the input data as a pandas DataFrame of features and labels."""
+  input_df = pd.read_csv(file_name, names=CSV_COLUMNS, skiprows=skiprows)
+
+  # replace missing values with np.nan and ignore
+  #input_df = input_df.replace([' ?'], [np.nan])
+
+  label_df = input_df.pop(LABEL_COL)
+  return (input_df, label_df)
+
+
+def generate_input(input_df, label_df):
+  """Prepare the input columns using SparseTensor."""
+  continuous_columns = [
+      tf.constant(input_df[col].values) for col in CONTINUOUS_COLS
+  ]
+
+  categorical_columns = [
+      tf.SparseTensor(
+          indices=[[i, 0] for i in range(input_df[col].size)],
+          values=input_df[col].astype('category').cat.codes.values,
+          dense_shape=[input_df[col].size, 1])
+      for col in CATEGORICAL_COLS
+  ]
+
+  sparse_t = tf.SparseTensor(
+      indices=[[i, 0] for i in range(label_df.size)],
+      values=label_df.astype('category').cat.codes.values,
+      dense_shape=[label_df.size, 1]
+  )
+
+  dense_i_tensor = tf.sparse_to_indicator(sparse_t, 2)
```

Comment:
Because I have to convert a [num_examples,] shape into [num_examples,2] shape. See 2 = number of classes.